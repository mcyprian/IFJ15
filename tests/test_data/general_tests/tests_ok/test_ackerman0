int ackerman(int m, int n)
{
    if(m) 
    {
        if(n) 
        {
            int x = n - 1;
            int y = m - 1;
            int v = ackerman(m, x);
            int b = ackerman(y,v);
            return b;
        }
        else
        { 
            m = m - 1; 
            int x = ackerman(m, 1); 
            return x;
        }

    }
    else
    { 
        return n + 1;
    }
    
}

int main()
{
    for(int m = 0; m <= 4; m = m +1)
    {
        for(int n = 0; n < 5 - m; n = n +1)
        {
            int a = ackerman(m,n);
            cout << "A " << m << " " << n << " " << a << "\n";
        }
    }

    return 42;
}
