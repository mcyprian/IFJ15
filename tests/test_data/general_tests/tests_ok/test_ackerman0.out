A 0 0 1
A 0 1 2
A 0 2 3
A 0 3 4
A 0 4 5
A 1 0 2
A 1 1 3
A 1 2 4
A 1 3 5
A 2 0 3
A 2 1 5
A 2 2 7
A 3 0 5
A 3 1 13
A 4 0 13
